var/spack/repos/builtin/packages/rocm-openmp-extras/package.py

```diff
@@ -149,11 +163,22 @@ def setup_build_environment(self, env):
 
     def patch(self):
         src = self.stage.source_path
+        flang_warning = '-Wno-incompatible-pointer-types-discards-qualifiers)'
         aomp_extras = '{0}/rocm-openmp-extras/aomp-extras/aomp-device-libs'
         libomptarget = \
             '{0}/rocm-openmp-extras/llvm-project/openmp/libomptarget'
         flang = '{0}/rocm-openmp-extras/flang/'
 
+        if self.spec.version < Version('4.1.0'):
+            plugin = '/plugins/hsa/CMakeLists.txt'
+        else:
+            # Spack thinks some warnings from the flang build are errors.
```

Comment:
What do you mean here?

Reply:
There were some warnings related to a header file that then print out the file's contents for reference:
flang/tools/flang1/include/errmsgdf.h:372:12: warning: initializing 'char *' with an expression of type 'const char [47]' discards qualifiers
/* 366 */ "Internal error: interprocedural analysis fails",

Spack thought this was an error and it is just a piece of the source code. Does Spack montior the build logs for certain signatures or phrases? For example, "error:"

In future released this warning should be fixed but for now I chose to ignore it.